File: core/src/main/scala/tajo/benchmark/Driver.scala
package tajo.benchmark

import java.lang.Class
import tajo.conf.TajoConf
;


object Driver {
  def printUsage() {
    println("benchmark BenchmarkClass datadir query")
  }

  def main(args : Array[String]) {
    if (args.length < 3) {
      printUsage();
      System.exit(-1);
    }
    val conf : TajoConf = new TajoConf()
    val clazz = Class.forName(args(0))
    val params = Array[AnyRef](conf, args(1))
    val benchmark = clazz.newInstance().asInstanceOf[BenchmarkSet]

    benchmark.init(conf, args(1))
    benchmark.loadSchemas()
    benchmark.loadQueries()
    benchmark.loadTables()
    benchmark.perform(args(2))
    System.exit(0);
  }
}

File: core/src/main/java/tajo/benchmark/Driver.java
package tajo.benchmark;

import tajo.conf.TajoConf;

public class Driver {
  private static void printUsage() {
    System.out.println("benchmark BenchmarkClass datadir query");
  }

  public static void main(String [] args) throws Exception {

    if (args.length < 3) {
      printUsage();
      System.exit(-1);
    }

    TajoConf conf = new TajoConf();
    Class clazz = Class.forName(args[0]);
    BenchmarkSet benchmark = (BenchmarkSet) clazz.newInstance();

    benchmark.init(conf, args[1]);
    benchmark.loadSchemas();
    benchmark.loadQueries();
    benchmark.loadTables();
    benchmark.perform(args[2]);
    System.exit(0);
  }
}
